EIPS/eip-4430.md

```diff
@@ -0,0 +1,147 @@
+
+This technique of blinded data will become much more popular with L2
+solutions, which use blinding not necessarily for privacy, but for 
+compression.
+
+### Entangling the Contract Address
+
+To prevent signed data being used across contracts, the contract address
+is entanlged into both the transaction implicitly via the `to` field.
+
+
+### Alternatives
+
+- NatSpec and company are a class of more complex languages that attempt to describe the encoded data directly. Because of the language complexity they often end up being quite large requiring entire runtime environments with ample processing power and memory, as well as additional sandboxing to reduce security concerns. One goal of this is to reduce the complexity to something that could execute on hardware wallets and other simple wallets. These also describe the data directly, which in many cases (such as blinded data), cannot adequately describe the data at all
+
+- Custom Languages; due to the complexity of Ethereum transactions, any language used would require a lot of expressiveness and re-inventing the wheel. The EVM already exists (it may not be ideal), but it is there and can handle everything necessary.
+
+- Format Strings (e.g. Trustless Signing UI Protocol; format strings can only operate on the class of regular languages, which in many cases is insufficient to describe an Ethereum transaction. This was an issue quite often during early attempts at solving this problem.
+
+- The signTypedData [EIP-712](./eip-712.md) has many parallels to what this EIP aims to solve
+
+
+## Backwards Compatibility
+
+This does not affect backwards compatibility.
+
+
+## Reference Implementation
+
+@TODO (consider adding it as one or more files in `../assets/eip-####/`)
+
+I will add examples in Solidity and JavaScript.
+
+
+## Security Considerations
+
+### Escaping Text
+
+Wallets must be careful when displaying text provided by contracts and proper
+efforts must be taken to sanitize it, for example, be sure to consider:
+
+- HTML could be embedded to attempt to trick web-based wallets into executing code using the script tag (possibly uploading any private keys to a server)
+- In general, extreme care must be used when rendering HTML; consider the ENS names `<span style="display:none">not-</span>ricmoo.eth` or `&thinsp;ricmoo.eth`, which if rendered without care would appear as `ricmoo.eth`, which it is not
+- Other marks which require escaping could be included, such as quotes (`"`), formatting (`\n` (new line), `\f` (form feed), `\t` (tab), any of many non-standard whitespaces), back-slassh (`\`)
+- UTF-8 has had bugs in the past which could allow arbitrary code execution and crashing renderers; consider using the UTF-8 replacement character (or *something*) for code-points outside common planes or common sub-sets within planes
+- Homoglyphs attacks
+- Right-to-left mark may affect rendering
+- Many other things, deplnding on your environment
+
```

Comment:
I'd recommend reiterating the concern the point made above that contracts can implement this maliciously (purposely giving an inaccurate description)